ouroboros-consensus/src/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -311,7 +311,14 @@ forkBlockProduction maxBlockBodySize IS{..} BlockProduction{..} =
       let withTxs :: (MempoolSnapshot blk TicketNo -> STM m a) -> m a
           withTxs = withSyncState mempool (TxsForBlockInSlot currentSlot)
 
+      -- We need to measure the trace just before we actually run the
+      -- protocol because it is possible that the the __current node
+      -- is not the slot leader__ and will __not lead__ the protocol/block
+      -- production.
+      -- Currently, we will simply drop the unmatched trace event and take
+      -- the diff between the last event of this type and the forged block event.
       trace $ TraceForgeAboutToLead currentSlot
+
```

Comment:
In fact, do we need this comment here? This seems like something that belongs in the monitoring/benchmarking code?

Reply:
Removed.